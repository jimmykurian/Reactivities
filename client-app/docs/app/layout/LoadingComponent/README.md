[**Client App Documentation v0.0.0**](../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../README.md) / app/layout/LoadingComponent

# app/layout/LoadingComponent

## Index

### Interfaces

- [Props](interfaces/Props.md)

### Functions

- [default](functions/default.md)
