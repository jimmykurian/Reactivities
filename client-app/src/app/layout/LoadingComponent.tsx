/**
 * @file LoadingComponent.tsx
 * @author Jimmy Kurian
 */

import { Dimmer, Loader } from 'semantic-ui-react';

/**
 * @interface Props
 * @description Props interface for the LoadingComponent.
 * @property {boolean} [inverted=true] - Whether the dimmer is inverted (dark background with light content).
 * @property {string} [content='Loading...'] - The loading message to be displayed.
 */
export interface Props {
  inverted?: boolean;
  content?: string;
}

/**
 * @component LoadingComponent
 * @description LoadingComponent renders a loading indicator with an optional message.
 * @param {Props} props - The properties passed to the component.
 * @returns {JSX.Element} The rendered LoadingComponent.
 * @remarks
 * This component utilizes the `Dimmer` and `Loader` components from the Semantic UI React library.
 * The `inverted` prop controls the background and content color scheme, and the `content` prop allows for a customizable loading message.
 * @example
 * ```tsx
 * <LoadingComponent inverted={false} content="Please wait..." />
 * ```
 */
export default function LoadingComponent({
  inverted = true,
  content = 'Loading...',
}: Props): JSX.Element {
  return (
    <Dimmer active={true} inverted={inverted} data-testid="dimmer">
      <Loader content={content} />
    </Dimmer>
  );
}
