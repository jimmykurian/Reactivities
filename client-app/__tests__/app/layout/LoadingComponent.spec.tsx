import { render, screen } from '@testing-library/react';
import LoadingComponent from '../../../src/app/layout/LoadingComponent';

describe('LoadingComponent', () => {
  test('renders with default props', () => {
    // Arrange & Act
    render(<LoadingComponent />);

    // Assert
    const loaderElement = screen.getByText('Loading...');
    expect(loaderElement).toBeInTheDocument();
  });

  test('renders with custom content', () => {
    // Arrange
    const customContent = 'Please wait...';

    // Act
    render(<LoadingComponent content={customContent} />);

    // Assert
    const loaderElement = screen.getByText(customContent);
    expect(loaderElement).toBeInTheDocument();
  });

  test('renders with inverted dimmer', () => {
    // Arrange & Act
    render(<LoadingComponent inverted={true} />);

    // Assert
    const dimmerElement = screen.getByTestId('dimmer');
    expect(dimmerElement).toHaveClass('inverted');
  });

  test('renders with non-inverted dimmer', () => {
    // Arrange & Act
    render(<LoadingComponent inverted={false} />);

    // Assert
    const dimmerElement = screen.getByTestId('dimmer');
    expect(dimmerElement).not.toHaveClass('inverted');
  });

  test('matches snapshot with default props', () => {
    // Arrange & Act
    const { asFragment } = render(<LoadingComponent />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });

  test('matches snapshot with custom content and non-inverted dimmer', () => {
    // Arrange
    const customContent = 'Loading data...';

    // Act
    const { asFragment } = render(
      <LoadingComponent content={customContent} inverted={false} />,
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
