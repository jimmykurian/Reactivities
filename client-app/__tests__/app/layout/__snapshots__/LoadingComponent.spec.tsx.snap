// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingComponent matches snapshot with custom content and non-inverted dimmer 1`] = `
<DocumentFragment>
  <div
    class="ui active transition visible dimmer"
    data-testid="dimmer"
    style="display: flex;"
  >
    <div
      class="content"
    >
      <div
        class="ui text loader"
      >
        Loading data...
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LoadingComponent matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="ui active transition visible inverted dimmer"
    data-testid="dimmer"
    style="display: flex;"
  >
    <div
      class="content"
    >
      <div
        class="ui text loader"
      >
        Loading...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
